### DISCUSSION TITLE
Open import functionality for developers

### DISCUSSION THREAD

**Discussion by codespren** [godotengine/godot-proposals#4021] [Engine Core]
Godot has good support for different filetypes and importing assets and changing their import settings is easy enough in the editor. However, this functionality is available only in the editor. If someone want to import assets from the code at runtime then they cannot use the same functionality but they need to create their own.

This can be very problematic with some file types, for example .wav-files. In practise, if someone wants to import .wav from the code they need to implement their own parser which is silly because Godot already contains such a parser but it is not accessable at runtime.

I don't know how the Godot's import functionality is organized but it would be great if it could be opened and used from the code.

**Comment by azur-wolve** [comment]
@codespren do you mean expose the related functions in the API?

**Reply by codespren** [reply]
That might work or perhaps more user-friendly solution would be to provide separate class that can be used to load resources at runtime. Similar to current ResourceLoader but it would work in exported projects too. Currently ResourceLoader works only when loading resources from res:// -folder.

**Comment by Xrayez** [comment]
See my proposal for this:
- #1632

**Comment by fire** [comment]
https://github.com/godotengine/godot/pull/47389 exists for wave? What are the list of runtime usable imports?
